### PR TITLE
Release 1.6.2

### LINKED ISSUE #1868: Polish currency formatting
When PLN currency is selected price format is: zł.200,00.
The correct format is: 200,00 zł.

### LINKED ISSUE #3100: Users should be able to "zoom" into a high-res image on the PDP
When browsing the PDP, users should be able to click, over or pinch to zoom on the currently active product image.

### LINKED ISSUE #3270: Better document OAuth methods
document https://github.com/reactioncommerce/reaction/pull/3231

### LINKED ISSUE #3274: Better document CollectionFS image settings
inspired by https://forums.reactioncommerce.com/t/product-images-sizes/616/8

### LINKED ISSUE #3314: After accessing a  product from the search page  , I am not able to continue scrolling down the single product page
When I hit the navbar search button , I select a product and  I am suppose to be able to continue scrolling down the single product page
but the page is completely stuck , I have to refresh my page to go back to normal 
**Steps to reproduce the behavior**
here is how you can reproduce the bug:
1) hit the search icon 
2) search for any product
3) select a product
4) try scrolling down the page
**Versions:**
Node: 7.7.1
NPM: 4.1.2
Meteor Node: 4.8.4
Meteor NPM: 4.6.1
Reaction CLI: 0.22.5
Reaction: 1.5.8

### LINKED ISSUE #3317: Tax is not shown in Completed Order page when using Avalara
### Expected behavior
Taxes should display as part of the order summary

### Actual behavior
They do not

### Steps to reproduce the behavior
1. Enable Avalara as Tax provider
1. Checkout as normal with an address that would be charged tax
1. Observe tax charged during checkout
1. Observe tax does not display on completed order page

### Versions
```
Node: 8.9.1
NPM: 5.5.1
Meteor Node: 4.8.4
Meteor NPM: 4.6.1
Reaction CLI: 0.22.5
Reaction: 1.5.8
Reaction branch: brent-fix-issue-2718 (master)
Docker: 17.09.0-ce
```

### LINKED ISSUE #3320: External websites not opening in a new tab.
### Expected behavior
Avalara website should open in a new tab.
### Actual behavior
Avalara website opens in the same tab.
### Steps to reproduce the behavior
1. Run `reaction`.
1. Open the tax menu from left sidebar.
1. Click on `Avalara Admin Console Login`, observe the site opening in the same tab.

### LINKED ISSUE #3328: Alavara tax being 0 case not handled.
### Expected behavior
No exception
### Actual behavior
Exception is thrown when `totalTax` returned by Alavara API is 0.

The `if` condition [here](https://github.com/reactioncommerce/reaction/blob/master/imports/plugins/included/taxes-avalara/server/hooks/hooks.js#L32) checks if `result.totalTax && typeof result.totalTax === "number"` which returns 0 if `result.totalTax === 0`, so the control goes [here](https://github.com/reactioncommerce/reaction/blob/master/imports/plugins/included/taxes-avalara/server/hooks/hooks.js#L38) to check `if (result.error.errorCode === 503)` which throws the exception as `result.error === undefined`

### LINKED ISSUE #3388: Cannot resend emails as an admin if email was successfully sent on the first try
### Expected behavior
As an admin I should be able to resend emails from the email panel.

### Actual behavior

As an admin, I cannot resend emails from the email panel. This is happening exclusively with emails that have already sent successfully once.

I can only resend emails that were unsuccessful on the first try.

### Steps to reproduce the behavior
- Do not setup email config
- Perform action that sends email, observe that it fails to send
- setup email config 
- resend email, see that it succefully sends
- on the same email try resend it, observe that it fails (status red)

### LINKED ISSUE #3402: Cannot show date picker filter in orders panel
### Expected behavior
Clicking the arrow to show the date filters in the orders panel should show a popover with a date picker.

### Actual behavior
The popover is not shown and an error appears in the console.

reaction

### Steps to reproduce the behavior

With all npm packages up to date on the `master` branch. Open the orders panel and try to reveal the date picker.

## Research

It seems like in the latest version of the `react-dates` package we need to initialize before use.

This is how the imports should look in `calendarPicker.js`.

```
import "react-dates/initialize";
import { DayPickerRangeController } from "react-dates";
```

While the above should allow the date picker to show up, **all** of the CSS is broken. Seems they refactored all of the class names. Seems most of them seem similar, but with removed hyphens and added camelCasing.

**summary**
- Need to initialize before use `import "react-dates/initialize";`
- Need to update all CSS in `calendarPicker.less`

### Versions

Node: 7.2.0
NPM: 5.3.0
Meteor Node: 8.8.1
Meteor NPM: 5.4.2
Reaction CLI: 0.24.2
Reaction: 1.6.1
Reaction branch: master
Docker: 17.09.0-ce

### LINKED ISSUE #3409: (docs): document constants in jsdoc
Need to document constants and the rest of the unorganized methods in api.docs.reactioncommerce.com

### PR DESCRIPTION
**Pull request by spencern**
# v1.6.2

## Potentially Breaking Changes
- (ref): remove unused Router code (#3412)
  **Note** This code could potentially break a downstream app or shop _if and only if_ it had added a publication which published all shops to the client and depended on this specific implementation in the router creating shop prefixed routes. This could have happened through a plugin or through direct modification of the core reaction code. Reaction core has not used these shop prefixed routes for at least several versions and it's unclear if these routes were ever used in core.
  With that said, please be diligent to test your routes if you update your shop and have made changes to the Shops publication or rely on specific shop-prefixed routes.
  From the PR
  > extraneous Router code that was left over from a previous refactor, and was causing some conflicts in the client project.
  > The code was used to push all routes from all shops into the Router. In core Reaction, this is unneeded functionality, as we do not publish all routes from all shops, therefore the code seemed to be doing nothing in it's current state.

## Fixes
 - (fix): Import PropTypes from "prop-types" package (#3417)
 - (fix): cart discount getting overridden to zero (#3414)
 - (fix): resend email from dashboard even if completed (#3410) - Resolves #3388
 - (fix): orders date filter not appearing (#3403) - Resolves #3402 
 - Using the default tax rate (#3326) - Resolves #3317 
 - Site will open in a new tab (#3327) - Resolves #3320 
 - Removing check for the tax being 0 (#3329) - Resolves #3328 
 - Polish currency formatting (#3330) - Resolves #1868
 - (fix): Determine denormalized product inventory status during Shopify import (#3366) - 
 - Fix: Not being able to scroll after using search function #3314 (#3315) - Resolves #3314

## Features
 - (feat): allow getShopName() to accept a shopId parameter (#3400)
 - (feat): zoom image in PDP (#3343) - Resolves #3100 

## Upgrades
-  Upgrade to Meteor v1.6.0.1 (#3407)

## Docs
- docs: document Accounts package helper methods (#3273) - Resolves #3270 
- docs(constants): #3409 document default variables (#3413) - Resolves #3409 
- doc: namespace server methods into Core (#3209)
- docs: document FS.Collection, Media (#3275) - Resolves #3274 
- update product screenshot to reflect new version (#3362)

## Contributors
Thanks to @Akarshit and @wackywombaat12 for contributing to this release

## Notes
There were additional updates that we were hoping to release at this time, #3367 and #3368. Those PRs contained potentially breaking changes. We have reverted those PRs and they are not included in this release. They will require more work on our Product Revisions module.

### CODE REVIEW

File: imports/plugins/core/accounts/client/helpers/accountsHelper.js [hunk 1]
```diff
@@ -32,8 +39,9 @@ export function sortGroups(groups) {
 }
 
 /**
- * getInvitableGroups - helper - client
- * @summary This generates a list of groups the user can invite to.
+ * @method getInvitableGroups
+ * @memberof Accounts
+ * @summary helper - client This generates a list of groups the user can invite to.
```
Comment (zenweasel) [inline imports/plugins/core/accounts/client/helpers/accountsHelper.js]: Typo here that makes it hard to understand what the function does? (inherited, but maybe we can fix it while we notice it?)

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "reaction",
   "description": "Reaction is a modern reactive, real-time event driven ecommerce platform.",
-  "version": "1.6.1",
+  "version": "1.7.0",
```
Comment (zenweasel) [inline package.json]: Should be 1.6.2?

File: server/api/core/core.js [hunk 3]
```diff
@@ -508,12 +672,20 @@ export default {
     Shops.update({ _id: shopId }, { $set: { layout: uniqLayouts } });
   },
 
+  /**
+   * @name getAppVersion
+   * @method
+   * @memberof Core
+   * @return {Number} App version
```
Comment (zenweasel) [inline server/api/core/core.js]: Incorrect: String

File: server/api/core/core.js [hunk 4]
```diff
@@ -790,18 +964,25 @@ export default {
       });
     });
   },
+
+  /**
+   * @name setAppVersion
+   * @method
+   * @memberof Core
+   * @return {Number} App version
```
Comment (zenweasel) [inline server/api/core/core.js]: Incorrect: Does not have a return value

File: package.json (hunk 2 above)
Comment (spencern) [reply package.json]: 👍

File: imports/plugins/core/accounts/client/helpers/accountsHelper.js (hunk 1 above)
Comment (spencern) [reply imports/plugins/core/accounts/client/helpers/accountsHelper.js]: @impactmass we may need to update the first line of this jsdoc summary to be a little easier to understand in the docs context

Comment (impactmass) [reply imports/plugins/core/accounts/client/helpers/accountsHelper.js]: Okay. I'll make a note of that for next PR

**Pull request merged**